pkg/controller/jobframework/reconciler.go

```diff
@@ -389,12 +389,14 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 			if err != nil {
 				log.Error(err, "Unsuspending job")
 				if podset.IsPermanent(err) {
+					allErrs := err
```

Comment:
wait a second.... If the error is permanent, we actually don't to return the error, otherwise that would cause a retry.

Reply:
Oops, It is a great catch :)